src/inline-assembly.md

```diff
@@ -1308,8 +1330,8 @@ r[asm.rules.arm64ec]
 - On arm64ec, [call checkers with appropriate thunks](https://learn.microsoft.com/en-us/windows/arm/arm64ec-abi#authoring-arm64ec-in-assembly) are mandatory when calling functions.
 
 r[asm.rules.only-on-exit]
-- The requirement of restoring the stack pointer and non-output registers to their original value only applies when exiting an `asm!` block.
-  - This means that `asm!` blocks that never return (even if not marked `noreturn`) don't need to preserve these registers.
+- The requirement of restoring the stack pointer and non-output registers to their original value only applies when exiting the assembly code.
+  - This means that assembly code that never return (even if not marked `noreturn`) don't need to preserve these registers.
```

Comment:
Most of the `asm!` block to assembly code changes need some grammar fixups. For example:
```suggestion
  - This means that assembly code that never returns (even if not marked `noreturn`) doesn't need to preserve these registers.
```